src/backend/distributed/commands/multi_copy.c

```diff
@@ -3571,6 +3613,52 @@ CopyGetPlacementConnection(HTAB *connectionStateHash, ShardPlacement *placement,
 	}
 
 	connection = GetPlacementConnection(connectionFlags, placement, nodeUser);
+	if (connection == NULL)
+	{
+		/*
+		 * For this COPY command, we have not established any connections and adaptive
```

Comment:
Maybe start this sentence with "If" or move it into the `if` block below, because at the moment it makes me think that the `connection == NULL` condition implies we have not established any connections, which is not quite true.